### PR TITLE
(#875) MinGW build support

### PR DESCRIPTION
**Pull request by The-Renaissance**
This pull request addresses issue #875.
I haven't figured out MinGW images for CI yet, so expect more commits once I got that figured out.
Also see issue #1008

### CODE REVIEW

**Comment by The-Renaissance** [comment]
wtf github... there's supposed to be a ninth commit

**Comment by The-Renaissance** [comment]
MinGW CI support is online!

File: README.md [hunk 1]
```diff
@@ -47,6 +47,10 @@ $ sudo pacman -S gcc cmake sdl2 inotify-tools libxml2
 - [Visual Studio 2015+](https://visualstudio.microsoft.com/)
 - [SDL2 VC Development Libraries](https://www.libsdl.org/release/SDL2-devel-2.0.9-VC.zip)
 
+### MinGW
```
Comment (rexim) [inline README.md]: MinGW should be a subsection of Windows. Also the Windows section should have a Visual Studio subsection for the current content.

File: README.md [hunk 2]
```diff
@@ -78,6 +82,37 @@ $ ../build/nothing
 > ..\build\nothing
 ```
 
+### MinGW (with MSYS)
```
Comment (rexim) [inline README.md]: Same situation here. Windows should have three subsections: "Visual Studio", "MinGW (with MSYS)" and "MinGW (without MSYS)"

**Comment by rexim** [comment]
@The-Renaissance please squash all of the commits into a single with a commit message that refers to the issue we are fixing #875

**Comment by The-Renaissance** [comment]
The changes you requested have been implemented. @rexim

**Comment by rexim** [comment]
@The-Renaissance thank you so much for the contribution! :+1:

**Pull request merged**